### PR TITLE
Quadriticization fixes

### PR DESCRIPTION
**Pull request by fabrizio-m**
Solves a few with quadraticization, also simplifies some code and removes generics from `fold_instance_witness_pair`, now it can be called with just inference.

### CODE REVIEW

**Comment by volhovm** [comment]
@fabrizio-m I'd suggest you add `cargo +nightly fmt` into your VSCode automatic settings (https://github.com/o1-labs/proof-systems/pull/2044), it should take <1min.

File: folding/src/quadraticization.rs [hunk 1]
```diff
@@ -90,11 +90,23 @@ fn lower_degree_to_1<C: FoldingConfig>(
     let degree = exp.degree();
     match degree {
         1 => exp,
-        _ => {
-            let exp = lower_degree_to_2(exp, rec);
-            let id = rec.get_id(exp);
-            FoldingExp::Atom(ExtendedFoldingColumn::WitnessExtended(id))
-        }
+        _ => match exp {
+            FoldingExp::Add(e1, e2) => FoldingExp::Add(
```
Comment (dannywillems) [inline folding/src/quadraticization.rs:94]: Do you have any example testing the output of quadraticization of simple expressions, like `X1 * X2`, `X1 * X2 * (X3 + X4)`, etc?

File: folding/src/expressions.rs [hunk 2]
```diff
@@ -56,6 +57,67 @@ pub enum FoldingCompatibleExpr<C: FoldingConfig> {
...
+                    c.to_string()
+                }
+                FoldingCompatibleExprInner::Challenge(c) => {
+                    format!("{:?}", c)
+                }
+                FoldingCompatibleExprInner::Cell(cell) => {
+                    let Variable { col, row } = cell;
+                    let next = match row {
+                        CurrOrNext::Curr => "",
+                        CurrOrNext::Next => " * ω",
```
Comment (mrmr1993) [inline folding/src/expressions.rs:79]: This is technically incorrect: we're getting `col(x)` for `Curr` and `col(x * \omega)` for `Next`. Might be easier to write simply `Curr` or `Next`.

**Review by dannywillems — APPROVED** [review]
Merging it, and see following PR which adds some documentations and maybe additional tests.

**Pull request merged**